sentinel/src/lib/feed.js

```diff
@@ -1,21 +1,81 @@
 const logger = require('./logger');
-const db = require('../db'),
-      tombstoneUtils = require('@medic/tombstone-utils');
+const db = require('../db');
+const metadata = require('./metadata');
+const tombstoneUtils = require('@medic/tombstone-utils');
 
 // we don't run transitions on ddocs or info docs
 const IDS_TO_IGNORE = /^_design\/|-info$/;
 
+const RETRY_TIMEOUT = 60000; // 1 minute
+
+let listener;
+let init;
+let request;
+
+const getProcessedSeq = () => {
+  return metadata
+    .getProcessedSeq()
+    .catch(err => {
+      logger.error('transitions: error fetching processed seq: %o', err);
+      return;
+    });
+};
+
+const registerFeed = seq => {
+  logger.info(`transitions: fetching changes feed, starting from ${seq}`);
+  request = db.medic
+    .changes({ live: true, since: seq })
+    .on('change', change => {
+      if (!change.id.match(IDS_TO_IGNORE) &&
+          !tombstoneUtils.isTombstoneId(change.id)) {
+        listener(change);
+      }
+    })
+    .on('error', err => {
+      logger.error('transitions: error from changes feed: %o', err);
+      init = null;
+      setTimeout(() => listen(), RETRY_TIMEOUT);
+    })
+    .catch(() => {
+      // catch to avoid unhandled rejection - it's handled above
+    });
```

Comment:
I looks like chaining the `catch` here messes up the return value, which ends up being an empty object. 
If I push incorrect configs, expecting the feed to be canceled, I get: 
```
2019-12-10 08:59:16 ERROR: Unknown transition "random" 
2019-12-10 08:59:16 ERROR: Transitions are disabled until the above configuration errors are fixed. 
2019-12-10 08:59:16 ERROR: Unhandled Rejection: 
2019-12-10 08:59:17 ERROR: TypeError: request.cancel is not a function
    at /home/diana/projects/medic/sentinel/src/lib/feed.js:73:17
    at processTicksAndRejections (internal/process/task_queues.js:89:5) {
  [stack]: 'TypeError: request.cancel is not a function\n    at ' +
    '/home/diana/projects/medic/sentinel/src/lib/feed.js:73:17\n   ' +
    ' at processTicksAndRejections ' +
    '(internal/process/task_queues.js:89:5)',
  [message]: 'request.cancel is not a function'
} 
```
Yet the feed is not canceled and keeps on advancing the seq as new changes are pushed. 
If I inspect `request` before calling `cancel()`, it's an empty object. Super weird. Might be because the `catch` is attached to a promise that never resolves when it's a live feed? Is it a PouchDB bug?
If I remove the chained `catch` and call it separately, `cancel` works fine.

Reply:
Removed. I must've changed the implementation at some point as I'm no longer getting the uncaught rejection error which makes this redundant anyway. Everything seems to be working now. Thanks for the tip!